[Fong Wei Jie] iP

WeiJie96 opened the pull request.

Commit:
Add Gradle support

WeiJie96 committed:
Chang Duke.java

- Add welcome, echo, exit

WeiJie96 committed:
Add, List
Added arraylist to ask tasks

WeiJie96 committed:
Mark as Done

WeiJie96 committed:
Todo, Event, Deadline
Added inheritance

WeiJie96 committed:
Exceptions
Handled 2 exceptions

WeiJie96 committed:
Delete
Handle delete, modify exception

WeiJie96 committed:
UITest

WeiJie96 committed:
Add javadoc

WeiJie96 committed:
Add enum, improve code quality, add javadoc

Handling tasks (deadline, todo, event) has common behaviour

Common behaviour cause code duplication
Extracting common behaviour also allows for exception handling

Let's gather the common lines of code into a new method
The method also handles exceptions

It is done this way to enhance readabilty and quality

In addition to the above, we also improve coding standard:
* Make line break before operator
* Make static variable private
* Make boolean variables boolean

WeiJie96 committed:
Add javadoc for non-trivial method and rename method names to verbs

WeiJie96 committed:
Add basic file manipulation

WeiJie96 committed:
Add space

WeiJie96 committed:
Add read write to disk functionality

Deadline, Event, Todo, Task classes's constructor do not take in
boolean isDone

Add overloaded constructor to all of them for convenience of file
reading

Let's create a file reader and file writer to read and save
the tasks in list, and also add overloaded constructors to classes
mentioned above

WeiJie96 committed:
Handle date and time in Deadline and Event

Deadline and Event dont understand date and time, treat as normal
string

This casuses potential issues later e.g. when we want to build
a calendar

Let's handle date times, and throw errors if the user does not
input in the right format

WeiJie96 committed:
Merge branch 'branch-Level-7'

WeiJie96 committed:
Merge branch 'branch-Level-8'

Resolve conflicts
# Conflicts:
#	src/main/java/Duke.java

WeiJie96 committed:
Use more OOP, by refactoring code

Input command by user has common behaviour e.g. need some sort
of saving and printing

Extracting common code allows for scalability and readability

Let's
* Pull out ui related code into one class
* Storage related code into one class
* Parsing realted code into one class
* Task activity into one class
* Command execution into respective classes

Using inheritance is preferred for the commands as they have
common actions and constructors

WeiJie96 committed:
Make variables private

WeiJie96 committed:
Add unit tests for Parser and TaskList

WeiJie96 committed:
Update gitignore

WeiJie96 committed:
Add Javadoc for more than half of methods/classes

WeiJie96 committed:
Implement coding standards

WeiJie96 committed:
Add functionality to find tasks based on exact match

WeiJie96 committed:
Merge branch 'branch-A-JavaDoc'

WeiJie96 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/AddCommand.java
#	src/main/java/ExitCommand.java
#	src/main/java/ListCommand.java
#	src/main/java/Parser.java

WeiJie96 committed:
Merge branch 'branch-Level-9'

WeiJie96 committed:
Dummy commit to tag A-Jar

WeiJie96 committed:
Add duke package to all files

WeiJie96 committed:
Merge branch 'add-gradle-support'

WeiJie96 committed:
Use gradle
Remove public access, change to package access for all methods
and classes

WeiJie96 committed:
Commit gradle

WeiJie96 committed:
Merge branch 'master' into branch-A-CheckStyle

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

WeiJie96 committed:
Update Checkstyle

WeiJie96 committed:
Merge branch 'branch-A-Gradle'

WeiJie96 committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/test/java/duke/ParserTest.java
#	src/test/java/duke/TaskListTest.java

ruixuantan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
```
Maybe can consider using a more descriptive name that indicates a date/time object?

ruixuantan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * Variable to store event time.
+     */
+    protected LocalDate at;
```
Perhaps can consider using a more descriptive name? I notice the same issue in the Deadline class.

ruixuantan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,95 @@
+     * @throws DukeException  When date is in wrong format, or task lacks description.
+     */
+    static Command parse(String input) throws DukeException {
+        String second;
```
Perhaps you can use something more descriptive like `secondCommand`?

ruixuantan reviewed (commented):
I genuinely like your code quality. Overall, the naming of classes and methods is good.
I also like the fact that your methods are doing only one thing (obeying SRP).

peter-yeh commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -0,0 +1,9 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+<<<<<<< HEAD
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
+=======
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.2-bin.zip
+>>>>>>> master
```
Is this merge conflict resolved?

peter-yeh commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,53 @@
+ */
+class AddCommand extends Command {
+
+    /** Name of command e.g. Todo. */
+    private CommandName commandName;
+    /** Description of task. */
+    private String description;
+    /** Date in string format. */
+    private String date;
```
I like how descriptions is given for these private variables as well

peter-yeh commented on src/main/java/duke/CommandName.java:
```diff
@@ -0,0 +1,8 @@
+/**
+ * Enumeration of all valid commands by user.
+ */
+enum CommandName {
```
I liked how enum is used for these commands

peter-yeh commented on src/main/java/duke/DoneCommand.java:
```diff
@@ -0,0 +1,43 @@
+     * @throws DukeException When I/O error occurs.
+     */
+    @Override
+    void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
```
I noticed a lot of your methods are marked as package private, why don't mark them as protected instead?

peter-yeh commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,56 @@
+     * @return Simple string description.
+     */
+    @Override
+    String toStringSimple() {
```
Could this method name be matched closer to what it does?

peter-yeh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+     * inner list is one line in file, split by delimiter
+     * @throws DukeException When IO error occurs.
+     */
+    ArrayList<ArrayList<String>> load() throws DukeException {
```
why is this not called after the constructor instead of calling it in duke.java?

peter-yeh reviewed (commented):
Overall, it was really hard to find any java coding convention errors. Minor questions were raised regarding some methods.

WeiJie96 committed:
Add GUI
Change all prints to return String, so that it can be passed
into XML elements for display

WeiJie96 committed:
Add welcome text
Close window after bye
Make overflow text visible, i.e. do not clip overflow text, e.g.
when list is too long
Change all response to remove indentation at the starting line
Remove unused functions in duke
Remove unsused functions in Ui

WeiJie96 committed:
Update gitignore

WeiJie96 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	data/tasks.txt
#	src/main/java/duke/AddCommand.java
#	src/main/java/duke/Command.java
#	src/main/java/duke/DeleteCommand.java
#	src/main/java/duke/DoneCommand.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/ExitCommand.java
#	src/main/java/duke/FindCommand.java
#	src/main/java/duke/ListCommand.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

WeiJie96 committed:
Resolve merge conflict for grdle wrapper properties

WeiJie96 committed:
Change the names of tasks in storage to enumerations

WeiJie96 committed:
Fix Ui bug of long text being clipped off (the proper fix)

WeiJie96 committed:
Add in assertions to if-else (else) and switch-case (default)

WeiJie96 committed:
Improve Code Quality
* Make variable names and method names inituitive
* Maintain SLAP
* Make magic strings into new (named) variables

WeiJie96 committed:
Merge pull request #1 from WeiJie96/branch-A-Assertions

Add in assertions to if-else (else) and switch-case (default)

WeiJie96 committed:
Merge branch 'master' into branch-A-Assertions

WeiJie96 committed:
Merge pull request #2 from WeiJie96/branch-A-CodeQuality

Improve Code Quality

WeiJie96 committed:
Merge branch 'master' into branch-A-CodeQuality

WeiJie96 committed:
Handle abstraction problem

TaskList handles both task management and returning ui message
This is not good in terms of abstraction in SRP.

Lets take out the ui messages and place them in UI class instead.

In addition
* Change images of duke and user
* Change welcome and bye message

WeiJie96 committed:
Maintain SLAP, and refactoring
- take out lines of code which handle the information passed from
add task and delete task to UI, and put into a new method

WeiJie96 committed:
Merge branch 'branch-A-CodeQuality'

WeiJie96 committed:
Merge branch 'branch-A-Assertions'

WeiJie96 committed:
Add basic list and add client functionality

WeiJie96 committed:
Merge branch 'branch-D-Clients'

WeiJie96 committed:
Revert "Merge branch 'branch-D-Clients'"

This reverts commit fb6d487919c4ad5e113b854aec09e4b6553d8587, reversing
changes made to 5668f3d150b2d73d1c4e90428c5a1b3fdb7d4a66.

WeiJie96 committed:
Implement sort by description

WeiJie96 committed:
Merge branch 'C-Sort'

WeiJie96 committed:
Handle more exceptions in user input.
Extract common code together.

There is common code for handling exceptions for indexing of array.

Let's pull out the common code, and put them into a single method.

This improves code readability, and also minimise code repetition.

WeiJie96 committed:
Add in more javadoc and comply to checkstyle.

WeiJie96 committed:
Add in more javadoc

WeiJie96 committed:
Handle error message better.

Error messages are created at the place they were found, e.g.
in parser, storage.

This should be the job of another class, and does not allow
for scalability and ease of change.

Let's pull out all the error messages, and create a new class to
handle them.

The use of enumerations also improves code readability and quality.

WeiJie96 committed:
* Change images to pokemon related
* Change error messages and ui messages to piplup related
* Style button, text, add button hover effect
* Add title to application

WeiJie96 committed:
Merge branch 'branch-A-BetterGui'

WeiJie96 committed:
Add in product screenshot

WeiJie96 committed:
Correct and add more javadoc.

WeiJie96 committed:
Improve code quality
* More intuitive variable names
* Splitting code into 2 separate methods

WeiJie96 committed:
Edit tests to make it pass

WeiJie96 committed:
* Add in User Guide
* Add in Image showcase
* Rename images

WeiJie96 committed:
Add readme to root folder

WeiJie96 committed:
Update readme in root folder

WeiJie96 committed:
Set theme jekyll-theme-cayman

WeiJie96 committed:
Set theme jekyll-theme-hacker

WeiJie96 committed:
Set theme jekyll-theme-cayman

WeiJie96 committed:
Remove alert info div tag in Readme

WeiJie96 committed:
Merge branch 'master' of https://github.com/WeiJie96/ip

WeiJie96 committed:
Delete theme from github pages as it is causing rendering issues

WeiJie96 committed:
Remove :information_source: in readme as it cannot be rendered in HTML

WeiJie96 committed:
Dummy commit to test GitHub Pages

WeiJie96 committed:
Dummy commit

WeiJie96 committed:
Remove readme

WeiJie96 committed:
Upload readme

WeiJie96 committed:
Add index file to test if site works
